src/client/contextChannels.ts

```diff
@@ -443,6 +464,24 @@ export async function getCurrentChannel(identity?: Identity): Promise<Channel> {
     return getChannelObject(channelTransport);
 }
 
+/**
+ * Returns an app channel with the given name. Either creates a new channel or returns an existing channel.
+ *
+ * It is up to your organization to decide how to share knowledge of these custom channels. It is recommended that
+ * channel names are prefixed with your organization's domain name to avoid name collisions.
+ *
+ * The service will assign a unique ID when creating a new app channel, but no particular mapping of name to ID should
+ * be assumed.
+ *
+ * @param name The name of the channel
+ */
+
+export async function getOrCreateAppChannel(name: string): Promise<AppChannel> {
```

Comment:
The idea is that, let's say FinTechCo have a suite of apps that all use "com.FinTechCo-taupe", "com.FinTechCo-magnolia", "com.FinTechCo-salmon", no single app can rely on another one of those apps having created the channel. It's whoever get there first that creates it, so we either provider `getOrCreate`, or the most common use-case of a seperate `get` and `create` is going to be emulating `getOrCreate`.

It might be too limiting eventually, but this has come out of Phil's convos with FactSet, and distinct `get`/`create`s are something we can always add if there's demaind

Reply:
Though realised typing this out, the comment should be "*reverse* domain name". Fixed

Reply:
That makes sense, in that case only a `hasAppChannel` would need to be added in future.